doc/server.nix

```
@@ -0,0 +1,24 @@
+{
+  docs,
+  http-server,
+  writeShellApplication,
+}:
+writeShellApplication {
+  name = "serve-docs";
+  runtimeInputs = [ http-server ];
+  runtimeEnv = {
+    server_flags = [
+      # Search for available port
+      "--port=0"
+
+      # Disable browser cache
+      "-c-1"
+
+      # Open using xdg-open
+      "-o"
+    ];
+  };
```

Nitpick:

```suggestion
  runtimeEnv.server_flags = [
    # Search for available port
    "--port=0"

    # Disable browser cache
    "-c-1"

    # Open using xdg-open
    "-o"
  ];
```

This is also the formatting you used in the NixVim patch: https://github.com/nix-community/nixvim/commit/28a2abf874c3ecbbf91edf1b2b9fe595f0f54099.